services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/controller/DebugVpcController.java

```diff
@@ -92,25 +91,26 @@ public Map getVpcCount() throws CacheException {
             method = POST,
             value = {"/debug/project/{projectid}/vpcs"})
     @ResponseStatus(HttpStatus.CREATED)
-    public VpcStateJson createVpcState(@PathVariable String projectid, @RequestBody VpcStateJson resource) throws Exception {
-        VpcState vpcState = null;
+    public VpcWebJson createVpcState(@PathVariable String projectid, @RequestBody VpcWebRequestJson resource) throws Exception {
+        VpcWebResponseObject inVpcState = new VpcWebResponseObject();
 
         try {
             RestPreconditionsUtil.verifyParameterNotNullorEmpty(projectid);
 
-            VpcState inVpcState = resource.getVpc();
+            VpcWebRequestObject vpcWebRequestObject = resource.getNetwork();
+            BeanUtils.copyProperties(vpcWebRequestObject, inVpcState);
             RestPreconditionsUtil.verifyResourceNotNull(inVpcState);
             RestPreconditionsUtil.populateResourceProjectId(inVpcState, projectid);
 
             Transaction transaction = this.vpcDatabaseService.getCache().getTransaction();
             transaction.start();
 
             this.vpcDatabaseService.addVpc(inVpcState);
-            vpcState = this.vpcDatabaseService.getByVpcId(inVpcState.getId());
+            inVpcState = this.vpcDatabaseService.getByVpcId(inVpcState.getId());
```

Comment:
This is the vpc state retrieved from database, meaning that it is not the "in"VpcState any more. Let us use a new variable number, like "persisted"VpcState, or "db"VpcState, or something else.